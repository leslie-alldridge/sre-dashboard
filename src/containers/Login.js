import React, { Component } from "react";
import { Redirect } from "react-router-dom";
import axios from "axios";

import TextField from "@material-ui/core/TextField";

import {
  LinkButtons,
  SubmitButtons,
  registerButton,
  homeButton,
  loginButton,
  forgotButton,
  inputStyle,
  HeaderBar
} from "../components";

const title = {
  pageTitle: "Login Screen"
};

class Login extends Component {
  constructor() {
    super();

    this.state = {
      username: "",
      password: "",
      loggedIn: false,
      showError: false,
      showNullError: false
    };
  }

  handleChange = name => event => {
    this.setState({
      [name]: event.target.value
    });
  };

  loginUser = e => {
    e.preventDefault();
    if (this.state.username === "" || this.state.password === "") {
      this.setState({
        showError: false,
        showNullError: true,
        loggedIn: false
      });
    } else {
      axios
        .post("/login", {
          username: this.state.username,
          password: this.state.password
        })
        .then(response => {
          if (
            response.data === "bad username" ||
            response.data === "passwords do not match"
          ) {
            this.setState({
              showError: true,
              showNullError: false
            });
          } else {
            localStorage.setItem("JWT", response.data.token);
            this.setState({
              loggedIn: true,
              showError: false,
              showENullrror: false
            });
          }
        })
        .catch(error => {
          console.log(error.data);
        });
    }
  };

  render() {
    const {
      username,
      password,
      showError,
      loggedIn,
      showNullError
    } = this.state;
    if (!loggedIn) {
      return (
        <div>
          <HeaderBar title={title} />
          <form className="profile-form" onSubmit={this.loginUser}>
            <TextField
              style={inputStyle}
              id="username"
              label="username"
              value={username}
              onChange={this.handleChange("username")}
              placeholder="Username"
            />
            <TextField
              style={inputStyle}
              id="password"
              label="password"
              value={password}
              onChange={this.handleChange("password")}
              placeholder="Password"
              type="password"
            />
            <SubmitButtons buttonStyle={loginButton} buttonText={"Login"} />
          </form>
          {showNullError && (
            <div>
              <p>The username or password cannot be null.</p>
            </div>
          )}
          {showError && (
            <div>
              <p>
                That username or password isn't recognized. Please try again or
                register now.
              </p>
              <LinkButtons
                buttonText={`Register`}
                buttonStyle={registerButton}
                link={"/register"}
              />
            </div>
          )}
          <LinkButtons
            buttonText={`Go Home`}
            buttonStyle={homeButton}
            link={"/"}
          />
          <LinkButtons
            buttonStyle={forgotButton}
            buttonText={"Forgot Password?"}
            link={"/forgotPassword"}
          />
        </div>
      );
    } else {
      return <Redirect to={`/userProfile/${username}`} />;
    }
  }
}
export default Login;
