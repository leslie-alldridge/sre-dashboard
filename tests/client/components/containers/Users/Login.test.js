import React from "react";
import { shallow } from "enzyme";
import Login from "../../../../../src/containers/Users/Login";
describe("MyComponent", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Login debug />);

    expect(component).toMatchSnapshot();
  });
});
