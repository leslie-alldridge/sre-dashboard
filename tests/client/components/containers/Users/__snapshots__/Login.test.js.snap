// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <HeaderBar
          title={
            Object {
              "pageTitle": "Login Screen",
            }
          }
        />,
        <form
          className="profile-form"
          onSubmit={[Function]}
        >
          <TextField
            id="username"
            label="username"
            onChange={[Function]}
            placeholder="Username"
            required={false}
            select={false}
            style={
              Object {
                "margin": ".5em",
              }
            }
            value=""
          />
          <TextField
            id="password"
            label="password"
            onChange={[Function]}
            placeholder="Password"
            required={false}
            select={false}
            style={
              Object {
                "margin": ".5em",
              }
            }
            type="password"
            value=""
          />
          <SubmitButtons
            buttonStyle={
              Object {
                "background": "royalblue",
                "margin": "1em",
                "padding": "1em",
              }
            }
            buttonText="Login"
          />
        </form>,
        false,
        false,
        <LinkButtons
          buttonStyle={
            Object {
              "background": "mediumpurple",
              "margin": "1em",
              "padding": "1em",
            }
          }
          buttonText="Go Home"
          link="/"
        />,
        <LinkButtons
          buttonStyle={
            Object {
              "background": "purple",
              "margin": "1em",
              "padding": "1em",
            }
          }
          buttonText="Forgot Password?"
          link="/forgotPassword"
        />,
      ],
      "id": "user-bg",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "title": Object {
            "pageTitle": "Login Screen",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <TextField
              id="username"
              label="username"
              onChange={[Function]}
              placeholder="Username"
              required={false}
              select={false}
              style={
                Object {
                  "margin": ".5em",
                }
              }
              value=""
            />,
            <TextField
              id="password"
              label="password"
              onChange={[Function]}
              placeholder="Password"
              required={false}
              select={false}
              style={
                Object {
                  "margin": ".5em",
                }
              }
              type="password"
              value=""
            />,
            <SubmitButtons
              buttonStyle={
                Object {
                  "background": "royalblue",
                  "margin": "1em",
                  "padding": "1em",
                }
              }
              buttonText="Login"
            />,
          ],
          "className": "profile-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "id": "username",
              "label": "username",
              "onChange": [Function],
              "placeholder": "Username",
              "required": false,
              "select": false,
              "style": Object {
                "margin": ".5em",
              },
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "id": "password",
              "label": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "required": false,
              "select": false,
              "style": Object {
                "margin": ".5em",
              },
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "buttonStyle": Object {
                "background": "royalblue",
                "margin": "1em",
                "padding": "1em",
              },
              "buttonText": "Login",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "form",
      },
      false,
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "buttonStyle": Object {
            "background": "mediumpurple",
            "margin": "1em",
            "padding": "1em",
          },
          "buttonText": "Go Home",
          "link": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "buttonStyle": Object {
            "background": "purple",
            "margin": "1em",
            "padding": "1em",
          },
          "buttonText": "Forgot Password?",
          "link": "/forgotPassword",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <HeaderBar
            title={
              Object {
                "pageTitle": "Login Screen",
              }
            }
          />,
          <form
            className="profile-form"
            onSubmit={[Function]}
          >
            <TextField
              id="username"
              label="username"
              onChange={[Function]}
              placeholder="Username"
              required={false}
              select={false}
              style={
                Object {
                  "margin": ".5em",
                }
              }
              value=""
            />
            <TextField
              id="password"
              label="password"
              onChange={[Function]}
              placeholder="Password"
              required={false}
              select={false}
              style={
                Object {
                  "margin": ".5em",
                }
              }
              type="password"
              value=""
            />
            <SubmitButtons
              buttonStyle={
                Object {
                  "background": "royalblue",
                  "margin": "1em",
                  "padding": "1em",
                }
              }
              buttonText="Login"
            />
          </form>,
          false,
          false,
          <LinkButtons
            buttonStyle={
              Object {
                "background": "mediumpurple",
                "margin": "1em",
                "padding": "1em",
              }
            }
            buttonText="Go Home"
            link="/"
          />,
          <LinkButtons
            buttonStyle={
              Object {
                "background": "purple",
                "margin": "1em",
                "padding": "1em",
              }
            }
            buttonText="Forgot Password?"
            link="/forgotPassword"
          />,
        ],
        "id": "user-bg",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "title": Object {
              "pageTitle": "Login Screen",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TextField
                id="username"
                label="username"
                onChange={[Function]}
                placeholder="Username"
                required={false}
                select={false}
                style={
                  Object {
                    "margin": ".5em",
                  }
                }
                value=""
              />,
              <TextField
                id="password"
                label="password"
                onChange={[Function]}
                placeholder="Password"
                required={false}
                select={false}
                style={
                  Object {
                    "margin": ".5em",
                  }
                }
                type="password"
                value=""
              />,
              <SubmitButtons
                buttonStyle={
                  Object {
                    "background": "royalblue",
                    "margin": "1em",
                    "padding": "1em",
                  }
                }
                buttonText="Login"
              />,
            ],
            "className": "profile-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "id": "username",
                "label": "username",
                "onChange": [Function],
                "placeholder": "Username",
                "required": false,
                "select": false,
                "style": Object {
                  "margin": ".5em",
                },
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "id": "password",
                "label": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "required": false,
                "select": false,
                "style": Object {
                  "margin": ".5em",
                },
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "buttonStyle": Object {
                  "background": "royalblue",
                  "margin": "1em",
                  "padding": "1em",
                },
                "buttonText": "Login",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "form",
        },
        false,
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "buttonStyle": Object {
              "background": "mediumpurple",
              "margin": "1em",
              "padding": "1em",
            },
            "buttonText": "Go Home",
            "link": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "buttonStyle": Object {
              "background": "purple",
              "margin": "1em",
              "padding": "1em",
            },
            "buttonText": "Forgot Password?",
            "link": "/forgotPassword",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
